### ISSUE TITLE
MPD doesn't index files in music_dir with question mark in filename

### ISSUE THREAD

**Issue by fenuks** [MusicPlayerDaemon/MPD#1325]
### Describe the bug
Files that have `?` in filename are not indexed by MPD. This problem wasn't present in MPD 0.21.26, that I has been using prior to an update to 0.23.2.

## Expected Behavior
Files with `?` should be indexed, as in older versions of MPD.

## Actual Behavior
Full rescans don't help. `mpc ls` doesn't show affected files. Removing `?` from filename, and updating database works as workaround. I have dozens of such files, because I have script that fetches podcasts, and uses title for filename (except `/` or when filename is too long).

## Version
```
Music Player Daemon 0.23.2 (v21.02.1)
Copyright 2003-2007 Warren Dukes <warren.dukes@gmail.com>
Copyright 2008-2021 Max Kellermann <max.kellermann@gmail.com>
This is free software; see the source for copying conditions.  There is NO
warranty; not even MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Database plugins:
 simple proxy upnp

Storage plugins:
 local curl


Decoders plugins:
 [vorbis] ogg oga
 [oggflac] ogg oga
 [flac] flac
 [opus] opus ogg oga
 [faad] aac
 [ffmpeg] 16sv 3g2 3gp 4xm 8svx aa3 aac ac3 adx afc aif aifc aiff al alaw amr anim apc ape asf atrac au aud avi avm2 avs bap bfi c93 cak cin cmv cpk daud dct divx dts dv dvd dxa eac3 film flac flc fli fll flx flv g726 gsm gxf iss m1v m2v m2t m2ts m4a m4b m4v mad mj2 mjpeg mjpg mka mkv mlp mm mmf mov mp+ mp1 mp2 mp3 mp4 mpc mpeg mpg mpga mpp mpu mve mvi mxf nc nsv nut nuv oga ogm ogv ogx oma ogg omg opus psp pva qcp qt r3d ra ram rl2 rm rmvb roq rpl rvc shn smk snd sol son spx str swf tak tgi tgq tgv thp ts tsp tta xa xvid uv uv2 vb vid vob voc vp6 vmd wav webm wma wmv wsaud wsvga wv wve
 [pcm]

Filters:


Tag plugins:
 id3tag

Output plugins:
 shout null fifo pipe alsa pulse httpd recorder

Encoder plugins:
 null opus wave flac

Input plugins:
 file alsa curl ffmpeg mms

Playlist plugins:
 extm3u m3u pls xspf asx rss soundcloud flac cue embcue

Protocols:
 file:// alsa:// ftp:// ftps:// http:// https:// mms:// mmsh:// mmst:// mmsu://

Other features:
 epoll iconv inotify ipv6 tcp un
```

**Comment by MaxKellermann** [comment]
Duplicate of https://github.com/MusicPlayerDaemon/MPD/issues/1316 - and fixed in 0.23.3
Always use the latest version before reporting a bug!

**Comment by fenuks** [comment]
Thanks, and sorry for a duplicate. Unfortunately, using newest version is not that easy, as I run MPD on OpenWRT.
